core/trino-main/src/main/java/io/trino/memory/ClusterMemoryManager.java

```diff
@@ -226,20 +225,21 @@ public synchronized void process(Iterable<QueryExecution> runningQueries, Suppli
 
         if (!lowMemoryKillers.isEmpty() && outOfMemory && !queryKilled) {
             if (isLastKillTargetGone()) {
-                callOomKiller(runningQueries);
+                callOomKiller(runningQueries, executionInfoSupplier);
```

Comment:
It appears to me that this might be changing what the state of the world the OOM killer may observe.

Before the changes, the OOM killer logic would run only after all other checks above. In the OOM situation, the `resourceOvercommit && outOfMemory` check above could select some query to be killed. Then `queryKilled==true` and OOM killer isn't invoked.

After the changes, we're not snapshotting the state of queries anymore and that state can be changing concurrently. This could potentially lead to a different final outcome.